src/volcanic_island_simulator.py

```diff
@@ -106,10 +161,13 @@ def update(self, dt):
         self.flow_router.run_one_step()
 
         # Apply fluvial erosion, transport, and deposition
+        self.space.run_one_step()
 
         # Switch boundaries back to full grid
+        self.grid.status_at_node[under_water] = self.grid.BC_NODE_IS_CORE
 
         # Apply submarine sediment transport
+        self.ssd.run_one_step(dt)
```

Comment:
all of your code related to adding the SimpleSubmarineDiffuser looks good to me!